lms/djangoapps/courseware/tests/test_entrance_exam.py

```diff
@@ -519,3 +569,116 @@ def test_skip_entrance_exame_gating(self):
         )
         for toc_section in self.expected_unlocked_toc:
             self.assertIn(toc_section, unlocked_toc)
+
+    def test_entrance_exam_gating_for_staff(self):
```

Comment:
`test_entrance_exam_gating` tests case when user is not staff. Added tests for course without entrance exam here might be redundant.